airflow-core/src/airflow/ui/src/hooks/useUrlBuilders.ts

```diff
@@ -0,0 +1,76 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useLocation, useParams } from "react-router-dom";
+
+import { buildDagRunUrl, buildTaskInstanceUrl, buildTaskUrl } from "src/utils/links";
+
+export const useTaskUrlBuilder = () => {
+  const { pathname } = useLocation();
+  const { dagId } = useParams<{ dagId: string }>();
+
+  return (params: { isGroup?: boolean; isMapped?: boolean; runId?: string; taskId: string }): string => {
+    const { isGroup = false, isMapped = false, runId, taskId } = params;
+
+    return buildTaskUrl({
+      currentPathname: pathname,
+      dagId: dagId ?? "",
+      isGroup,
+      isMapped,
+      runId,
+      taskId,
+    });
+  };
+};
+
+export const useDagRunUrlBuilder = () => {
+  const { pathname } = useLocation();
+  const { dagId } = useParams<{ dagId: string }>();
+
+  return (runId: string): string =>
+    buildDagRunUrl({
+      currentPathname: pathname,
+      dagId: dagId ?? "",
+      runId,
+    });
+};
+
+export const useTaskInstanceUrlBuilder = () => {
+  const { pathname } = useLocation();
+  const { dagId } = useParams<{ dagId: string }>();
+
+  return (params: {
+    isGroup?: boolean;
+    isMapped?: boolean;
+    mapIndex?: string;
+    runId: string;
+    taskId: string;
+  }): string => {
+    const { isGroup = false, isMapped = false, mapIndex, runId, taskId } = params;
+
```

Comment:
This pattern seems odd. (maybe I missed something).

We allow passing `isMapped`, `runId`, `taskId` to the returned function. But those are actually overridden by the query params? Why do we allow to path them in the first place if those are not used? 

Same for bellow hooks